### PR TITLE
[#12] RPM support

### PR DESCRIPTION
**Pull request by GuChad369**
- Add a new `.spec` file to make the project support RPM installation.
- Create `RPM.md` to provide instructions.

I am not familiar with RPM, and although it seems to work well, there may still be some issues. 

I am not confident about this. Please review and let me know.

Thanks!

### CODE REVIEW

File: pgmoneta_ext.spec [hunk 1]
```diff
@@ -0,0 +1,86 @@
+Name:          pgmoneta_ext
+Version:       0.1.0
+Release:       1%{dist}
+Summary:       An extension to enhance the functionality of the main pgmoneta project.
+License:       BSD
+URL:           https://github.com/pgmoneta/pgmoneta_ext
+Source0:       https://github.com/pgmoneta/pgmoneta_ext/archive/%{version}.tar.gz
+
+BuildRequires: git gcc cmake make postgresql-server postgresql-server-devel
+Requires:      check check-devel
```
Comment (jesperpedersen) [inline pgmoneta_ext.spec]: `Requires:` is run-time requirements, so `postgresql-server`

File: pgmoneta_ext.spec [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+%description
+pgmoneta_ext is an extension to enhance the functionality of the main pgmoneta project.
+
+%define _debugsource_template %{nil}
```
Comment (jesperpedersen) [inline pgmoneta_ext.spec]: You don't need this

File: pgmoneta_ext.spec [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+cd build
+cmake -DCMAKE_BUILD_TYPE=Release ..
+%{__make}
+
+%install
+
+cd build
+make DESTDIR=%{buildroot} install
+
+%{__mkdir} -p %{buildroot}%{_docdir}/%{name}/manual
```
Comment (jesperpedersen) [inline pgmoneta_ext.spec]: We don't do the manual as part of the .rpm

File: pgmoneta_ext.spec [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+%{__install} -m 644 %{_builddir}/%{name}-%{version}/LICENSE %{buildroot}%{_docdir}/%{name}/LICENSE
+%{__install} -m 644 %{_builddir}/%{name}-%{version}/CODE_OF_CONDUCT.md %{buildroot}%{_docdir}/%{name}/CODE_OF_CONDUCT.md
+%{__install} -m 644 %{_builddir}/%{name}-%{version}/README.md %{buildroot}%{_docdir}/%{name}/README.md
+%{__install} -m 644 %{_builddir}/%{name}-%{version}/doc/DEVELOPERS.md %{buildroot}%{_docdir}/%{name}/DEVELOPERS.md
+%{__install} -m 644 %{_builddir}/%{name}-%{version}/doc/PR_GUIDE.md %{buildroot}%{_docdir}/%{name}/PR_GUIDE.md
+%{__install} -m 644 %{_builddir}/%{name}-%{version}/doc/RPM.md %{buildroot}%{_docdir}/%{name}/RPM.md
+%{__install} -m 644 %{_builddir}/%{name}-%{version}/doc/TEST.md %{buildroot}%{_docdir}/%{name}/TEST.md
+
+%{__install} -m 644 %{_builddir}/%{name}-%{version}/doc/manual/01-introduction.md %{buildroot}%{_docdir}/%{name}/manual/01-introduction.md
```
Comment (jesperpedersen) [inline pgmoneta_ext.spec]: Remove these

File: pgmoneta_ext.spec [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+%files
+%license %{_docdir}/%{name}/LICENSE
+%{_docdir}/%{name}/CODE_OF_CONDUCT.md
+%{_docdir}/%{name}/DEVELOPERS.md
+%{_docdir}/%{name}/PR_GUIDE.md
+%{_docdir}/%{name}/README.md
+%{_docdir}/%{name}/RPM.md
+%{_docdir}/%{name}/TEST.md
+%{_docdir}/%{name}/manual/01-introduction.md
```
Comment (jesperpedersen) [inline pgmoneta_ext.spec]: Remove these

File: pgmoneta_ext.spec [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+%{_docdir}/%{name}/manual/dev-02-git.md 
+%{_docdir}/%{name}/manual/dev-03-test.md 
+%{_docdir}/%{name}/manual/dev-04-rpm.md 
+%{_docdir}/%{name}/manual/user-00-head.md 
+%{_docdir}/%{name}/manual/user-01-quickstart.md 
+%{_docdir}/%{name}/manual/user-02-functions.md
+
+%{_libdir}/pgsql/pgmoneta_ext.so
+%{_libdir}/pgsql/pgmoneta_ext.so.0
+%{_libdir}/pgsql/pgmoneta_ext.so.0.1.0
```
Comment (jesperpedersen) [inline pgmoneta_ext.spec]: Use `0*`

File: pgmoneta_ext.spec [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+%{_datadir}/pgsql/extension/pgmoneta_ext--0.1.0.sql
+%{_datadir}/pgsql/extension/pgmoneta_ext.control
+
+/usr/src/debug/pgmoneta_ext-0.1.0-1.fc38.aarch64/src/include/utils.h
```
Comment (jesperpedersen) [inline pgmoneta_ext.spec]: Remove these

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by jesperpedersen** [comment]
See the `.spec` file for pgmoneta/core for more examples

**Comment by GuChad369** [comment]
Thank you for the advice. I have updated the request accordingly.

I retested it, and it works well.

**Comment by jesperpedersen** [comment]
Merged.

Thanks for your contribution !

**Pull request merged**